colossus/src/main/scala/colossus/core/Worker.scala

```diff
@@ -470,11 +396,18 @@ private[colossus] class Worker(config: WorkerConfig) extends Actor with ActorMet
           con.handleConnected()
         } catch {
           case c: java.net.ConnectException => {
+            //TODO consider rolling in connection failure as a disconnect cause
```

Comment:
Yeah I can eliminate the duplicate code, but I do want to maintain that behavior, since `ConnectException` is normal but anything else in that situation would be something really unexpected and should be noted.